students/CCSt130/lesson03/mail_room_pt2.py

```diff
@@ -0,0 +1,413 @@
+
+        # Prompt user for input
+        input_msg = "Please type the name of an Organization: "
+        # Capture user choice 
+        a_donor = str(input(input_msg))
+        print()
+        # Could put a loop here to allow correction
+        print("\nYou entered: {}".format(a_donor))
+
+        # Strip white space for comparison
+        input_strip = a_donor.replace(' ', '')
+
+        if a_donor in list_opt:
+            # Display list again per reqs
+            list_donor_names(all_donations)
+
+        # Filter out non-alpha entry            
+        elif input_strip.isalpha():
+            break 
+        else:
+            print()                
+            print("Invalid entry--try again!")
+            print()
+
+    # New donation option
+    print()
+    print("Would you like to enter a new Donation?")
+
+    while True:
+        input_msg = "Please choose 'Y' for Yes and 'N' for No. [Y,N]: "
+        # Capture user choice 
+        usr_input = str(input(input_msg))
+        
+        if usr_input == 'Y' or usr_input == 'y': # Why, yes!
+            break
+        elif usr_input == 'N' or usr_input == 'n': # Well, no.
+            print()
+            print(">>> No Donation Added. <<<")
+            return(a_donor)
+        elif usr_input == 'Q' or usr_input == 'q': # Outta here
+            return
+        elif usr_input.isdigit(): 
+            print("Invalid entry--try again!") # A for effort
+            print()
+        else: 
+            print("Invalid entry--try again!")
+            print()
+
+    # Assign date to a name
+    dt_today = (date.today())
+    # Necessary to convert to a string 
+    date_str = str(dt_today)    
+    
+    # Include donor name already entered
+    if a_donor != None:
+        print()
+        print("A Donation from '{}' will be entered into our list...".format(a_donor))
+    # Prompt for donation amount
+    while True:
+        amt_input = (input("Enter amount of donation without separators (e.g. '12333444'): "))   
+        # Check for numeric entry
+        if amt_input.isdigit():
+            new_amt = float(amt_input)
+            # Error handling should unrealistic amts be entered
+            if new_amt > 0 and new_amt < 1000000000: # n < billion
+                # append to dict
+                all_donations[a_donor][date_str] = new_amt
+                # Replaces all entries for existing donor
+                # all_donations[a_donor] = {dt: new_amt}
+                break
+            else:
+                print()
+                print("Amount must be greater than zero and less than a billion--try again!")
+                print()
+        else: # Non-numeric entry
+            print()
+            print("Invalid entry--try again!") 
+            print()
+    # Inform user
+    print()
+    print(">>> Donation Has Been Entered for {}. <<<".format(a_donor))
+    print()
+    print("###")
+    
+    return(a_donor) # Return user's choice
```

Comment:
This function breaks when adding new donors and their donations, do you know why?